Ch06/ch06/accounts/AccountManager.java

```diff
@@ -11,50 +11,41 @@ public class AccountManager {
 
     private final BlockingQueue<TransferTask> pending = new LinkedBlockingQueue<>();
     private final BlockingQueue<TransferTask> forDeposit = new LinkedBlockingQueue<>();
-    private final BlockingQueue<TransferTask> failed = new LinkedBlockingQueue<>();
+    private final BlockingQueue<TransferTask> failedInsufficientFunds = new LinkedBlockingQueue<>();
 
     private Thread withdrawals;
     private Thread deposits;
 
     public void init() {
         Runnable withdraw = () -> {
-            LOOP:
-            while (!shutdown) {
+            boolean interrupted = false;
+            while (!interrupted || !pending.isEmpty()) {
```

Comment:
This loop condition doesn't make much sense to me. As long as interrupted is false OR pending is empty, the loop will continue. So even after shutdown is set to true the loop will continue and the thread will not end. So, if somehow a task is added to the pending queue after shutdown, as described in the book, it will get processed, however the thread will not exit, as the loop is not conditional on shutdown.